ui/plugins/pinboard.js

```diff
@@ -74,26 +74,30 @@ treeherder.controller('PinboardCtrl', [
         };
 
         $scope.save = function() {
-            if ($scope.user.loggedin) {
-                if ($scope.enteringBugNumber) {
-                    // we should save this for the user, as they likely
-                    // just forgot to hit enter.
-                    $scope.saveEnteredBugNumber();
+            if($scope.canSaveClassifications()) {
```

Comment:
You could do both here, as a notification.  Have these two ``if``s independent and do a notify of either, both, or none.  And then after that, check both before allowing it.  feels a little redundant, I know, but those checks are pretty fast, I think.  And nice to get both errors out there if both exist.  I really hate that feeling of fixing one thing, try again, fail again, fix another thing....  

Something like:

    if (!$scope.user.loggedin) {
        thNotify...
    }
    if (!$scope.canSaveClassifications()) {
        thNotify...
    }
    if ($scope.canSaveClassifications() && $scope.user.loggedin) {
        save it...
    }